### ISSUE TITLE
SaveOnExitFile does not work for quit;

### ISSUE THREAD

**Issue by kiryph** [gap-system/gap#5035]
The documentation for [`SaveOnExitFile`](https://docs.gap-system.org/doc/ref/chap6.html#X843C07A4869EAA1D) says:

> If, when GAP is exiting due to a quit or end-of-file (i.e. not due to a QUIT) the variable `SaveOnExitFile` is bound to a string value, then the system will try to save the GAP workspace to that file, see `SaveWorkspace` ([3.3-1](https://docs.gap-system.org/doc/ref/chap3.html#X876544A57C73C488)).

### Configuration

gaprc:
```
❯ cat $HOME/Library/Preferences/GAP/gaprc # macOS
SaveOnExitFile := "/Users/kiryph/.cache/gap/workspacefile";
```

zsh alias in .zshrc:
```
❯ type gap
gap is an alias for /usr/local/bin/gap -L /Users/kiryph/.cache/gap/workspacefile
```

### Observed behaviour

```
❯ gap
gap> L:=10;
gap> quit; # or ctrl-d

❯ gap # restart gap
gap> L;
Error, Variable: 'L' must have a value
not in any function at *stdin*:1
```

### Expected behaviour

```
❯ gap
gap> L:=10;
gap> quit; # or ctrl-d

❯ gap # restart gap
gap> L;
10
```

### Copy and paste GAP banner (to tell us about your setup)
```
 ┌───────┐   GAP 4.12.0 of 2022-08-18
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-apple-darwin20-default64-kv8
 Configuration:  gmp 6.2.1, GASMAN, readline
 Loaded workspace: /Users/kiryph/.cache/gap/workspacefile
 Packages:   AClib 1.3.2, Alnuth 3.2.1, AtlasRep 2.1.4, AutoDoc 2022.07.10, AutPGrp 1.11, Browse 1.8.14, CaratInterface 2.3.4, CRISP 1.4.5,
             crypting 0.10, Cryst 4.1.25, CrystCat 1.1.10, CTblLib 1.3.4, curlInterface 2.2.3, FactInt 1.6.3, FGA 1.4.0, Forms 1.2.8, GAPDoc 1.6.6,
             genss 1.6.7, IO 4.7.2, IRREDSOL 1.4.3, json 2.1.0, LAGUNA 3.9.5, Memoisation 1.0, orb 4.8.5, Polenta 1.3.10, Polycyclic 2.16, PrimGrp 3.4.2,
             RadiRoot 2.9, recog 1.3.2, ResClasses 4.7.3, SmallGrp 1.5, Sophus 1.27, SpinSym 1.5.2, TomLib 1.2.9, TransGrp 3.6.3, utils 0.76
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'
```

### My Fix: Add modified function from `lib/session.g` to own gaprc

https://github.com/gap-system/gap/blob/1c11fb166967fcf0e3ad5a78ee9258743f3e7782/lib/session.g#L20-L25

New gaprc:
```gap
InstallAtExit( function()
    if IsBound(SaveOnExitFile)  and
       IsString(SaveOnExitFile) then
        SaveWorkspace(SaveOnExitFile);
    fi;
end);

SaveOnExitFile := "/Users/kiryph/.cache/gap/workspacefile";
```

Apparently, the `QUITTING` condition is always triggered not only for `QUIT` or `ForceQuit([ret]);`

**Comment by fingolfin** [comment]
Huh, weird, it works fine over here (also on macOS):
```
$ gap-4.12 -b
gap> SaveOnExitFile:="foo.wsp";;
gap> L:=10;
10
gap> quit;
$ ls foo.wsp
foo.wsp
$ gap-4.12 -b -L foo.wsp
gap> L;
10
```

The check for `QUITTING` is used to distinguish `quit` from `QUIT`. It is unclear to me why your proposed change would have any effect in your scenario. Odd.

**Comment by fingolfin** [comment]
Thank you. This is used to work and accidentally regressed in GAP 4.11 due to commit 49603c7c9b7cd34520da9e0f291da2ff0fac9de1 from PR #2856 (CC @ChrisJefferson). We'll fix it in GAP 4.12.1.

**Comment by fingolfin** [comment]
See PR #5043
